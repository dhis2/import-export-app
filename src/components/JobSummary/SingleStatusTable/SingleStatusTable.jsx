import i18n from '@dhis2/d2-i18n'
import {
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
    TableRowHead,
    TableCellHead,
} from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'

const SingleStatusTable = ({ status, description }) => (
    <Table>
        <TableHead>
            <TableRowHead>
                <TableCellHead>{i18n.t('Status')}</TableCellHead>
                <TableCellHead>{i18n.t('Description')}</TableCellHead>
            </TableRowHead>
        </TableHead>
        <TableBody>
            <TableRow>
                <TableCell>{status}</TableCell>
                <TableCell>{description}</TableCell>
            </TableRow>
        </TableBody>
    </Table>
)

SingleStatusTable.propTypes = {
    description: PropTypes.string,
    status: PropTypes.string,
}

export { SingleStatusTable }
