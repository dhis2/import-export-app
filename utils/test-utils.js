import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
