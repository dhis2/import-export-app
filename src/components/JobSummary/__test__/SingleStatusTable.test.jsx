import React from 'react'
import { render } from 'test-utils'

import { SingleStatusTable } from '../SingleStatusTable/SingleStatusTable.jsx'

const props = {
    status: 'WARNING',
    description: 'Import process completed successfully',
}

it(`matches snapshot`, () => {
    const { asFragment } = render(<SingleStatusTable {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
