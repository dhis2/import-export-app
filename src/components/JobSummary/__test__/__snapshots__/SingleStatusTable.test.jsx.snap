// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="jsx-2430604489 "
    data-test="dhis2-uicore-table"
  >
    <thead
      data-test="dhis2-uicore-tablehead"
    >
      <tr
        class="jsx-3124144591 zebraStriping"
        data-test="dhis2-uicore-tablerowhead"
      >
        <th
          class="jsx-1796049005 "
          data-test="dhis2-uicore-tablecellhead"
        >
          Status
        </th>
        <th
          class="jsx-1796049005 "
          data-test="dhis2-uicore-tablecellhead"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody
      data-test="dhis2-uicore-tablebody"
    >
      <tr
        class="jsx-3124144591 zebraStriping"
        data-test="dhis2-uicore-tablerow"
      >
        <td
          class="jsx-996330601 "
          data-test="dhis2-uicore-tablecell"
        >
          WARNING
        </td>
        <td
          class="jsx-996330601 "
          data-test="dhis2-uicore-tablecell"
        >
          Import process completed successfully
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
